superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -1511,9 +1511,19 @@ export function createDatasourceFailed(err) {
 export function createDatasource(vizOptions) {
   return dispatch => {
     dispatch(createDatasourceStarted());
+    const { dbId, schema, datasourceName, sql } = vizOptions;
     return SupersetClient.post({
-      endpoint: '/superset/sqllab_viz/',
-      postPayload: { data: vizOptions },
+      endpoint: '/api/v1/dataset/',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        database: dbId,
+        schema: schema,
+        table_name: datasourceName,
+        sql: sql,
+        owners: [1],
```

Comment:
the `owners` key can probably just be omitted from the payload here, then the backend should default to making the current user owner I believe